SupportFiles/jira_plugin-install.sh

```diff
@@ -0,0 +1,157 @@
+#!/bin/bash
+#
+# Script to download and install JIRA Plugins from other sources
+#
+#################################################################
+# shellcheck disable=SC2086
+PROGNAME="$(basename ${0})"
+SCRIPTHOME="${HOME:-/root}"
+
+JIRADCHOME="/var/atlassian"
+JIRAINSTDIR="/opt/atlassian/jira"
+DBCFGDIR="${JIRADCHOME}/application-data/jira"
+PLUGINS1DIR="${JIRAINSTDIR}/atlassian-jira/WEB-INF/lib"
+PLUGINS2DIR="${DBCFGDIR}/plugins"
+INSTLPLUGINS2DIR="${DBCFGDIR}/plugins/installed-plugins"
+
+# Array of URLs to download plugins (Type 1 or 2)
+PLUGINS1JARURL=(
+                 # <http or https link to jar file>
+               )
+PLUGINS2JARURL=(
+                 # <http or https link to jar or obr file>
+               )
+
+# Install Unzip if needed
+unzip -v && echo "Unzip already installed" || \
+  yum install unzip -y
+
+# Misc error-handler
+function err_exit {
+   local ERRSTR="${1}"
+   local SCRIPTEXIT=${2:-1}
+
+   # Our output channels
+   # echo "${ERRSTR}" > /dev/stderr
+   logger -t "${PROGNAME}" -p kern.crit "${ERRSTR}"
+
+   # Need our exit to be an integer
+   if [[ ${SCRIPTEXIT} =~ ^[0-9]+$ ]]
+   then
+      exit "${SCRIPTEXIT}"
+   else
+      exit 1
+   fi
+}
+
+# Make git run quietly...
+quiet_git() {
+   if [[ $( git "$@" < /dev/null > /dev/null 2>&1 )$? -eq 0 ]]
+   then
+      echo "Git-fetch successful"
+   else
+      err_exit "Git-fetch failed"
+   fi
+}
+
+# Create git staging-area as needed
+if [[ -d ${SCRIPTHOME}/git ]]
+then
+   echo "Git stagining-area already exists"
+else
+   printf "Creating central location for Git-hosted resources... "
+   # shellcheck disable=SC2015
+   install -d -m 000700 ${SCRIPTHOME}/git && echo "Success" || \
+     err_exit "Failed creating git staging-area."
+fi
+
+#########################################
+##                                     ##
+##    Plugins Type 1 Installation      ##
+##                                     ##
+#########################################
+
+# Check if JIRA Plugins 1 folder exists, create if necessary
+if [[ -d ${PLUGINS1DIR} ]]
+then
+  echo "${PLUGINS1DIR} exists - skipping create"
+else
+  printf "Creating %s... " "${PLUGINS1DIR}"
+  install -d -o root -g root -m 0755 "$PLUGINS1DIR" && echo "Success!" ||
+    err_exit "Failed to create ${PLUGINS1DIR}"
+fi
+
+if [[ ${#PLUGINS1JARURL[@]} -ne 0 ]]
+then
+  # Pull down plugins
+  for URL in "${PLUGINS1JARURL[@]}"
+  do
+    printf "Pulling down %s... " "${URL}"
+    curl -skL "${URL}" > "${PLUGINS1DIR}/${URL##*/}" && echo "Success!" || \
```

Comment:
Some consumers of this script might want to host the script in services like S3 or Artifactory.
- With the whole "best practices default bucket policy is 'private'" thing, you'll probably want to verify that you actually downloaded a file and not a `permission denied` document. Otherwise, you can end up with a false-success, here, that will cause explosions in later steps (or cause plugins to be silently discarded)
- Related: Ensure that script documentation clearly denotes that binaries hosted in an S3 bucket will need to be set either `--acl=public-read` or that the user will need to create pre-signed URLs to the S3-hosted plugins
- For authenticated URLS:
    - You'll want to test that using a `https://<USER>:<PASS>@<FQDN>/PATH/TO/FILE` works reliably`
    - Ensure documentation for script spells out that they have to use `https://<USER>:<PASS>@<FQDN>/PATH/TO/FILE` style URLs